File: src/main/java/rbq/wtf/lycoris/client/module/Modules/Movement/KeepSprint.java
package rbq.wtf.lycoris.client.module.Modules.Movement;


import rbq.wtf.lycoris.client.event.api.EventTarget;
import rbq.wtf.lycoris.client.event.events.EventMotionUpdate;
import rbq.wtf.lycoris.client.module.Module;
import rbq.wtf.lycoris.client.module.ModuleCategory;
import rbq.wtf.lycoris.client.utils.MovementUtils;
import rbq.wtf.lycoris.client.value.BooleanValue;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.potion.Potion;

public class KeepSprint extends Module {

    public final BooleanValue noDelay = new BooleanValue("NoDelay", true, this);
    public final BooleanValue allDirections = new BooleanValue("All Directions", false, this);
    public final BooleanValue blindness = new BooleanValue("Blindness Check", true, this);
    public final BooleanValue food = new BooleanValue("FoodStats Check", true, this);

    public KeepSprint() {
        super("KeepSprint", ModuleCategory.Movement, 0);
    }

    @Override
    public void onEnable() {

    }

    @Override
    public void onDisable() {

    }


    // 应当使用EventMove，但是暂时还没想好该事件如何实现
    @EventTarget
    public void EventMotionUpdate(EventMotionUpdate e) {
        if (e.isPre()) {
            if (!MovementUtils.isMoving() || mc.getPlayer().isSneaking() ||
                    (blindness.get() && mc.getPlayer().getEntityPlayerInstance().getEntityLivingBaseInstance().isPotionActive(Potion.getBlindness())) ||
                    (food.get() && !(mc.getPlayer().getEntityPlayerInstance().getFoodStats().getFoodLevel() > 6.0F || mc.getPlayer().getEntityPlayerInstance().getCapabilities().isAllowFlying()))
                    /*|| (checkServerSide.get() && (mc.thePlayer.onGround || !checkServerSideGround.get())
                    && !allDirectionsValue.get() && RotationUtils.targetRotation != null &&
                    RotationUtils.getRotationDifference(new Rotation(mc.thePlayer.rotationYaw, mc.thePlayer.rotationPitch)) > 30)*/) {
                mc.getPlayer().setSprinting(false);
                //Logger.debug("Set Sprint false", "SprintModule");
            } else if (allDirections.get() || mc.getPlayer().getMovementInput().getMoveForward() >= 0.8F) {
                mc.getPlayer().setSprinting(true);
                //Logger.debug("Set Sprint", "SprintModule");
            }
            if (noDelay.get()) {
                mc.getPlayer().setSprintingTicksLeft(0);
            }
        }
    }
}

File: src/main/java/rbq/wtf/lycoris/client/module/modules/movement/KeepSprint.kt
package rbq.wtf.lycoris.client.module.modules.movement

import rbq.wtf.lycoris.client.event.EventTarget
import rbq.wtf.lycoris.client.event.UpdateEvent
import rbq.wtf.lycoris.client.module.Module
import rbq.wtf.lycoris.client.module.ModuleCategory
import rbq.wtf.lycoris.client.module.ModuleInfo
import rbq.wtf.lycoris.client.utils.MovementUtils
import rbq.wtf.lycoris.client.value.BooleanValue
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.potion.Potion


@ModuleInfo(
    name = "KeepSprint",
    description = "Automatically sprints all the time.",
    category = ModuleCategory.Movement
)
class KeepSprint : Module() {
    private val noDelay = BooleanValue("NoDelay", true)

    //private val allDirections = BooleanValue("All Directions", false)
    private val blindness = BooleanValue("Blindness Check", true)
    private val food = BooleanValue("FoodStats Check", true)

    private val checkServerSide = BooleanValue("CheckServerSide", false)
    private val checkServerSideGround = BooleanValue("CheckServerSideOnlyGround", false)

    @EventTarget
    fun onUpdate(event: UpdateEvent) {
        if (!MovementUtils.isMoving || mc.player.isSneaking || blindness.get() && mc.player.isPotionActive(Potion.blindness) || food.get() && !(mc.player.foodStats.foodLevel > 6.0f
                    || mc.player.capabilities.isAllowFlying) || (checkServerSide.get() && (mc.player.isOnGround || !checkServerSideGround.get())
                    /*&& !allDirections.get()*/) /*&& RotationUtils.targetRotation != null && RotationUtils.getRotationDifference(Rotation(mc.player.rotationYaw, mc.player.rotationPitch)) > 30 */
        ) {
            mc.player.setSprinting(false)
        } else if (/*allDirections.get() || */mc.player.movementInput.moveForward >= 0.8f) mc.player.setSprinting(true)
        if (noDelay.get()) mc.player.setSprintingTicksLeft(0)
    }
}